charts/core/src/Grid/Grid.tsx

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+
+import { useChartContext } from '../ChartContext';
+
+import { useGridOptions } from './config';
+import { GridProps } from './Grid.types';
+
+export function Grid({ horizontal = true, vertical = true }: GridProps) {
+  const { updateChartOptions } = useChartContext();
+  const gridOptions = useGridOptions({ horizontal, vertical });
+
+  useEffect(() => {
+    updateChartOptions(gridOptions);
+  }, [gridOptions]);
```

Comment:
We're listening to a state that we control here. That's a signal that we're using `useEffect` wrong.

I'd recommend either updating the chart options from within `useGridOptions`, or just move the content of `useGridOptions` into this component

Reply:
I like the idea of just getting rid of the `useGridOptions` hook entirely. With this component's sole responsibility being to update options, I don't think it will bloat the component in any way, and brings the logic closer to the component. Updating!